### ISSUE TITLE
Bug: Resource metadata may not suppress resources with crafted comments in YAML templates

### ISSUE THREAD

**Issue by lingrlongr** [aws-cloudformation/cfn-python-lint#1691]
#### Version

Latest (0.35.1)

#### Description of issue

There's an interesting case where when using a YAML CloudFormation document, suppression of a rule using resource metadata does not work.  Inserting a comment at the top of the resource triggers this problem.

This is a sample template snippet that attempts to suppress a custom rule, `E9999`:

```
Resources:
  MyKey:
    # trouble comment
    Type: AWS::KMS::Key
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E9999
    Properties:
      # snip
```

Providing you actually have a rule `E9999`, you'll still see the failure for this rule.  If you take out that `# trouble comment` line, the error will go away.  It seems to be caused by behavior seen in [template.py](https://github.com/aws-cloudformation/cfn-python-lint/blob/93659fba5cf86ee34177c55b28df0aa9140b7dc9/src/cfnlint/runner.py#L67).  To be more explicit, the comparison will be something like:

```
if directive.get('start') <= match.linenumber <= directive.get('end'):
					
   ^ 19                      ^ 18                ^ 169
```

... where the difference between `directive.get('start')` and `match.linenumber` is the same number as the number of comment lines at the top.

**Comment by PatMyron** [comment]
I actually was not able to reproduce this with non-custom rules:
```yaml
Resources:
  Resource:
    # comment
    Type: AWS::KMS
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3001
```
```yaml
Resources:
  Resource:
    # comment
    Type: AWS::KMS::Key
    Properties:
      Property: Property
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3002
            - E3003
```
```yaml
Resources:
  Resource:
    # comment
    Type: AWS::KMS::Key
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3002
            - E3003
    Properties:
      Property: Property
```

[Is there a chance the custom rule match line number is actually flagging the top-level `Resources` instead of the specific resource?](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/docs/getting_started/rules.md#add-a-rulematch)

**Comment by kddejong** [comment]
I think I see where this can be improved. Out of curiosity does your custom rule flag the path for the issue as?  Is it `Resources/MyKey`?

**Comment by PatMyron** [comment]
@kddejong reproduced with:
```yaml
Resources:
  Resource:
    # comment
    Type: AWS::KMS::Key
    Metadata:
      cfn-lint:
        config:
          ignore_checks:
            - E3001
```

**Comment by lingrlongr** [comment]
> I think I see where this can be improved. Out of curiosity does your custom rule flag the path for the issue as? Is it `Resources/MyKey`?

Yes, it does.

**Comment by kddejong** [comment]
@lingrlongr can you check version v0.36.0 that was just released? This should be fixed and we can re-open if it isn't.
